src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const result = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete newState[key];
+        });
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

Comment:
According to the checklist, a `switch` statement should always have a `default` case for error handling. This helps to catch any unexpected `action.type` values and handle them appropriately.